invisible_cities/cities/detsim_waveforms.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+
+##################################
+######### WAVEFORMS ##############
+##################################
+def bincounter(xs : np.array,
+               dx : float =1.,
+               x0 : float =0.):
+    ixs = ((xs - x0) // dx).astype(int)
+    return np.unique(ixs, return_counts=True)
+
+
+def create_waveform(times : np.array,
+                    pes   : np.array,
+                    bins  : np.array,
+                    wf_bin_time : float,
+                    nsamples    : int) -> np.array:
+    wf = np.zeros(len(bins))
+
+    if np.sum(pes)==0:
+        return wf
+    if nsamples==0:
+        sel = pes>0
+        t = np.repeat(times[sel], pes[sel])
+        t = np.clip  (t, 0, bins[-1])
```

Comment:
Are you sure you want to clip like that? stuff that wouldn't fall in the buffer are going to fall in the edges.